### PR TITLE
Improvement: Improving the way to extract arguments while raising errors

### PR DESCRIPTION
**Pull request by shubhamzanwar**
image

testcase summary for reference

### CODE REVIEW

File: lib/index.js [hunk 1]
```diff
@@ -104,23 +104,18 @@ internals.implementation = function(server, options) {
...
-      }
     }
+    // Since it is clearly speacified in the docs that
+    // the errorFunc must return an object with keys:
+    // errorType and message, we need not worry about
+    // errorContext being undefined
+    const _errorType = errorContext.errorType;
+    const _message = errorContext.message;
+    const _scheme = errorContext.scheme;
+    const _attributes = errorContext.attributes;
```
Comment (nelsonic) [inline lib/index.js]: Why define four `const`? if we're trying to improve/optimise this 
could we simply inline them in the `return` statement and avoid using up memory?

File: lib/index.js [hunk 2]
```diff
@@ -104,23 +104,18 @@ internals.implementation = function(server, options) {
...
-        _message = errorContext.message;
-        _scheme = errorContext.scheme;
-        _attributes = errorContext.attributes;
-      }
     }
+    // Since it is clearly speacified in the docs that
+    // the errorFunc must return an object with keys:
+    // errorType and message, we need not worry about
+    // errorContext being undefined
+    const _errorType = errorContext.errorType;
```
Comment (nelsonic) [inline lib/index.js]: only the `_errorType` is required, the remaining 3 `const`s can be in-lined. 👍

File: lib/index.js [hunk 3]
```diff
@@ -104,23 +104,18 @@ internals.implementation = function(server, options) {
       scheme: scheme,
       attributes: attributes,
     };
-    let _errorType = errorType; // copies of params
-    let _message = message; // so we can over-write them below
-    let _scheme = scheme; // without a linter warning
-    let _attributes = attributes; // if you know a better way please PR!
 
-    if (options.errorFunc && internals.isFunction(options.errorFunc)) {
+    if (internals.isFunction(options.errorFunc)) {
```
Comment (nelsonic) [inline lib/index.js]: The idea behind having this check is that we test for presence of the key before testing for type.
But you're _right_, the `internals.isFunction` does this check already, so OK to remove. 👍

**Comment by shubhamzanwar** [comment]
@nelsonic I've updated the PR as per your comments.
I've removed all four declared constants in order to maintain consistency. I'm using them directly while calling the Boom error.

Also, the documentation mentions that both errorType and message are required for the options.errorFunc function:
image

Is this something we should update in the docs?

**Pull request merged**